<?xml version="1.0"?>

<launch>

  <arg name="line_type" default="inner"/>

  <!-- launch the dynamic reconfigure GUI -->
  <node pkg="rqt_reconfigure" name="rqt_console" type="rqt_reconfigure" args="-t"/>

  <!-- launch the yellow line detection node -->
  <node pkg="shifted_line_pkg" name="yellow_line" type="yellow_line.py" required="true" output="screen"/>

  <!-- launch the control unit node -->
  <node pkg="shifted_line_pkg" name="control_unit" type="control_unit.py" required="true" output="screen"/>

  <!-- launch the line follow node -->
  <node pkg="shifted_line_pkg" name="$(arg line_type)_line_follow" type="$(arg line_type)_line_follow.py" required="true" output="screen"/>

  <!-- launch the sdt report node -->
  <node pkg="shifted_line_pkg" name="sdt_report" type="sdt_report.py" required="true" output="screen"/>

</launch>
